### ISSUE TITLE
Guide for noobs?

### ISSUE THREAD

**Issue by theRustyKnife** [gate-sso/gate#17]
I'm trying to setup gate, but I'm an utter noob at these kinds of things and am getting stuck. Could I get some 'noobs guide' on how to set this up with g suite?

As for some concrete questions:

- What do I need to install prior to installing gate? Mysql? Redis? Anything else?
- What do the environmental variables mean? The description is quite ambiguous for someone who barely understands how oauth works and #16 confuses me even more. Also "hosted domain for gmail" and "domain for email address" sound like the same thing to me...
- How can I set this up with a different oauth provider? I don't want to start with our 'production' g suite right away.
- At least a link to a tutorial on how to setup a rails app with nginx/puma would be nice. All of the ones I found ended with something like "Point puma to your app and you're done", which is exactly the part I don't know how to do...

Additionally, I found that `bundle install` will fail on `mysql` with ruby 2.4, but works fine with 2.3.3. Is that intended?

I'm sorry to bother with such stupid questions, but I really like the idea of gate and it seems to be the only thing around allowing this type of setup. The documentation is just way too scarce for me to get it working. I could only find the stuff here on GitHub and then the blog post [here](https://blog.gojekengineering.com/how-vpn-automation-ended-up-becoming-a-single-sign-on-server-part-1-244e2ee2df09). Maybe I missed something...

I'm using Ubuntu Server 16.04 LTS if that matters.

Thanks

**Comment by lanejlanej** [comment]
This will help you: https://www.digitalocean.com/community/tutorials/how-to-deploy-a-rails-app-with-puma-and-nginx-on-ubuntu-14-04

**Comment by rgarrigue** [comment]
Hi

I'm far from being a noob, but I must say the doc is a bit lacking. 

- I wish I could have a couple of screenshots in the README.md, to get a better idea about what is Gate.
- I wish I could git clone + docker-compose up + http://localhost:3000 + login admin/password to get started. Tried, but docker-compose is asking for .env, and there's no explanation about what you put in there. I guess passwords, secrets, configuration to use another oauth source... but I can only guess the format.

Right now, I'ld like to know how to plug to an oauth source, namely my nextcloud ?

**Comment by ajeygore** [comment]
Hello,

If you set environment variables like this

export GATE_DB_HOST=localhost
export GATE_DB_PORT=3306
export GATE_DB_USER=<gate_db_user>
export GATE_DB_PASSWORD=<db_password>
export GATE_HOSTED_DOMAINS="your domains"

export JRUBY_OPTS="--dev"

We use Jruby 9.1.6.0

also, we only support google auth strings, but it can work on anything I suppose.

Let me know if you are facing any more issues.

**Comment by waldman** [comment]
What about this guys here:

```GATE_VPN_SSL_PVTKEY        - Private key for signing vpn mobileconfig
GATE_VPN_SSL_CERT          - SSL key for signing vpn mobileconfig
GATE_VPN_SSL_XSIGNED       - Cross signed key for signing vpn mobileconfig```

Are they the certificates generated by OpenVPN on install time? What's this Cross Signed key? Who signs it? me?

**Comment by ajeygore** [comment]
Yes. user does, we have included Figaro now, it should be better now. I am closing this issue.
